src/metabase/native_query_analyzer.clj

```diff
@@ -0,0 +1,55 @@
+(ns metabase.native-query-analyzer
+  "Integration with Macaw, which parses native SQL queries. All SQL-specific logic is in Macaw, the purpose of this
+  namespace is to:
+
+  1. Translate Metabase-isms into generic SQL that Macaw can understand.
+  2. Contain Metabase-specific business logic."
+  (:require
+   [clojure.string :as str]
+   [macaw.core :as mac]
+   [metabase.util :as u]
+   [toucan2.core :as t2]))
+
+(defn- normalize-name
+  ;; TODO: This is wildly naive and will be revisited once the rest of the plumbing is sorted out
+  [name]
+  (-> name
+      (str/replace "\"" "")
+      u/lower-case-en))
+
+(defn- fields-for-query
+  [q db-id]
+  (let [{:keys [columns tables]} (-> q
+                                     mac/parsed-query
+                                     mac/query->components)]
+    (t2/query {:select [[:f/id :field_id] [:f/name :column_name] [:t/name :table_name]]
+               :from [[(t2/table-name :model/Field) :f]]
+               :join [[(t2/table-name :model/Table) :t] [:= :table_id :t.id]]
+               :where [:and
+                       [:= :t.db_id db-id]
+                       [:in :%lower.t/name (map normalize-name tables)]
+                       [:in :%lower.f/name (map normalize-name columns)]]})))
```

Comment:
Does this handle the case where the columns where namespaced with a table alias in the query?